solc/CommandLineParser.cpp

```diff
@@ -840,13 +840,20 @@ bool CommandLineParser::processArgs()
 	else
 		m_options.input.mode = InputMode::Compiler;
 
-	if (
-		m_args.count(g_strExperimentalViaIR) > 0 &&
-		m_options.input.mode != InputMode::Compiler &&
-		m_options.input.mode != InputMode::CompilerWithASTImport
-	)
+	map<string, set<InputMode>> validOptionInputModeCombinations = {
+		{g_strErrorRecovery, {InputMode::Compiler, InputMode::CompilerWithASTImport}},
```

Comment:
Please add a comment to make it clear that this is not complete yet.
```suggestion
	map<string, set<InputMode>> validOptionInputModeCombinations = {
		// TODO: This should eventually contain all options.
		{g_strErrorRecovery, {InputMode::Compiler, InputMode::CompilerWithASTImport}},
```